core/src/main/scala/za/co/absa/spline/harvester/QueryExecutionEventHandler.scala

```diff
@@ -29,32 +26,14 @@ class QueryExecutionEventHandler(
   harvesterFactory: LineageHarvesterFactory,
   lineageDispatcher: LineageDispatcher) {
 
-  /**
-   * The method is executed when an action execution is successful.
-   *
-   * @param funcName   A name of the executed action.
-   * @param qe         A Spark object holding lineage information (logical, optimized, physical plan)
-   * @param durationNs Duration of the action execution [nanoseconds]
-   */
-  def onSuccess(funcName: String, qe: QueryExecution, durationNs: Long): Unit = {
+  def handle(qe: QueryExecution, result: Try[Duration]): Unit = {
```

Comment:
- It seems to me the main result here is the QueryExecution not a duration (maybe the result should be renamed to duration)
- The Try seems to suggest the error happened when generating the duration, not durring the whole job execution

Maybe in this case `Either` would be better or two params instead of one Try?

Reply:
done